semver/parse_range_test.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright Isaac Z. Schlueter and Contributors. All rights reserved. ISC license.
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import { assert } from "../assert/mod.ts";
+import { parseRange } from "./parse_range.ts";
+import type { Comparator, Range } from "./types.ts";
+
+function equalComp(
+  c0: Comparator,
+  c1: Comparator,
+): boolean {
+  return (
+    c0.operator === c1.operator &&
+    c0.major === c1.major &&
+    c0.minor === c1.minor &&
+    c0.patch === c1.patch
+  );
+}
+
+function equalComps(
+  cs0: Comparator[],
+  cs1: Comparator[],
+): boolean {
+  if (cs0.length !== cs1.length) {
+    return false;
+  }
+  for (let i = 0; i < cs0.length; i++) {
+    if (!equalComp(cs0[i], cs1[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function equalRanges(
```

Comment:
I don't think we need this util. We can use `assertEquals` to compare nested objects.